pkg/tsdb/postgres/postgres_test.go

```diff
@@ -15,12 +15,98 @@ import (
 	"github.com/grafana/grafana/pkg/services/sqlstore/sqlutil"
 	"github.com/grafana/grafana/pkg/tsdb"
 	"github.com/grafana/grafana/pkg/tsdb/sqleng"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"xorm.io/xorm"
 
 	_ "github.com/lib/pq"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
+// Test newPostgresQueryEndpoint.
+func TestNewPostgresQueryEndpoint(t *testing.T) {
+	origFactory := sqleng.NewXormEngine
+	t.Cleanup(func() {
+		sqleng.NewXormEngine = origFactory
+	})
+	var gotDriver string
+	var gotConnStr string
+	sqleng.NewXormEngine = func(driverName string, connStr string) (*xorm.Engine, error) {
+		gotDriver = driverName
+		gotConnStr = connStr
+		return nil, fmt.Errorf("stop here")
+	}
+
+	testCases := []struct {
```

Comment:
A test with SSL params would be nice so we cover that part of connection string generation while touching this part of the codebase.

Reply:
@bergquist Done.